docs/pt/docs/advanced/custom-response.md

```diff
@@ -0,0 +1,344 @@
+
+///
+
+## Utilizando `ORJSONResponse`
+
+Por exemplo, se você precisa bastante de performance, você pode instalar e utilizar o <a href="https://github.com/ijl/orjson" class="external-link" target="_blank">`orjson`</a> e definir a resposta para ser uma `ORJSONResponse`.
+
+Importe a classe, ou subclasse, de `Response` que você deseja utilizar e declare ela no *decorador de operação de rota*.
+
+Para respostas grandes, retornar uma `Response` diretamente é muito mais rápido que retornar um dicionário.
+
+Isso ocorre por que, por padrão, o FastAPI irá verificar cada item dentro do dicionário e garantir que ele seja serializável para JSON, utilizando o mesmo[Codificador Compatível com JSON](../tutorial/encoder.md){.internal-link target=_blank} explicado no tutorial. Isso permite que você retorne **objetos abstratos**, como modelos do banco de dados, por exemplo.
+
+Mas se você tem certeza que o conteúdo que você está retornando é **serializável com JSON**, você pode passá-lo diretamente para a classe de resposta e evitar o trabalho extra que o FastAPI teria ao passar o conteúdo pelo `jsonable_encoder` antes de passar para a classe de resposta.
+
+```Python hl_lines="2  7"
+{!../../docs_src/custom_response/tutorial001b.py!}
+```
+
+/// info | Informação
+
+O parâmetro `response_class` também será usado para definir o "media type" da resposta.
+
+Neste caso, o cabeçalho HTTP `Content-Type` irá ser definido como `application/json`.
+
+E será documentado como tal no OpenAPI.
+
+///
+
+/// tip | Dica
+
+A `ORJSONResponse` está disponível apenas no FastAPI, e não no Starlette.
+
+///
+
+## Resposta HTML
+
+Para retornar uma resposta com HTML diretamente do **FastAPI**, utilize `HTMLResponse`.
+
+* Importe `HTMLResponse`
+* Passe `HTMLResponse` como o parâmetro de `response_class` do seu *decorador de operação de rota*.
+
+```Python hl_lines="2  7"
+{!../../docs_src/custom_response/tutorial002.py!}
+```
+
+/// info | Informação
+
+O parâmetro `response_class` também será usado para definir o "media type" da resposta.
+
+Neste caso, o cabeçalho HTTP `Content-Type` será definido como `text/html`.
+
+E será documentado como tal no OpenAPI.
+
+///
+
+### Retornando uma `Response`
+
+Como visto em [Retornando uma Resposta Diretamente](response-directly.md){.internal-link target=_blank}, você também pode sobrescrever a resposta diretamente na sua *operação de rota*, ao retornar ela.
+
+O mesmo exemplo de antes, retornando uma `HTMLResponse`, poderia parecer com:
+
+```Python hl_lines="2  7  19"
+{!../../docs_src/custom_response/tutorial003.py!}
+```
+
+/// warning | Aviso
+
+Uma `Response` retornada diretamente em sua *função de operação de rota* não será documentada no OpenAPI (por exemplo, o `Content-Type` não será documentado) e não será visível na documentação interativa automática.
+
+///
+
+/// info | Informação
+
+Obviamente, o cabeçalho `Content-Type`, o código de status, etc, irão vir do objeto `Response` que você retornou.
```

Comment:
I think that "*irão vir*" is a little strange here. Maybe "*virão* do objeto" is better